lib/Analysis/Utility.cpp

```diff
@@ -271,8 +271,9 @@ unsigned ScanLoweringHelper::getAxisBlockStride() {
   for (unsigned dim : order) {
     if (dim == getAxis())
       return stride;
-    stride *= type.getShape()[dim] /
-              (sizePerThreads[dim] * threadsPerWarp[dim] * warpsPerCTA[dim]);
+    stride *= std::max<unsigned int>(
+        1, type.getShape()[dim] /
+               (sizePerThreads[dim] * threadsPerWarp[dim] * warpsPerCTA[dim]));
```

Comment:
Do you think this change could be upstreamed?
If not, could you wrap it into `#ifdef USE_ROCM`, so it will be highlighted on IFU?

Reply:
Let me poke upstream people to see what they think.

Reply:
Upstream PR is merged: https://github.com/openai/triton/pull/2273